src/Discord.Net.WebSocket/Entities/Messages/SocketUserMessage.cs

```diff
@@ -147,7 +147,18 @@ public Task ModifySuppressionAsync(bool suppressEmbeds, RequestOptions options =
         public string Resolve(TagHandling userHandling = TagHandling.Name, TagHandling channelHandling = TagHandling.Name,
             TagHandling roleHandling = TagHandling.Name, TagHandling everyoneHandling = TagHandling.Ignore, TagHandling emojiHandling = TagHandling.Name)
             => MentionUtils.Resolve(this, 0, userHandling, channelHandling, roleHandling, everyoneHandling, emojiHandling);
-        
+
+        /// <inheritdoc />
+        public async Task CrosspostAsync(RequestOptions options = null)
+        {
+            if (!(Channel is SocketNewsChannel))
+            {
+                throw new InvalidOperationException("Publishing (crossposting) is only valid in news channels.");
```

Comment:
Ditto above.

Reply:
Thanks for the guidance / assistance. Tidied up all feedback.